### PR TITLE
Upgrade to Stretch

### PR DESCRIPTION
**Pull request by jiru**
This PR is a rework based on #37 (without commits unrelated to Stretch (specific config for envy and sloth, nginx vhost for tatoeba.org, repackage of nihongoparser, sinoparser, suggestd and tatodetect)). It fixes things that were still broken and removes most of Ansible warnings about deprecated stuff.

### CODE REVIEW

File: ansible/roles/install_sphinx/tasks/main.yml [hunk 1]
```diff
@@ -21,10 +21,3 @@
   with_items:
     - /etc/init.d/sphinxsearch
     - /etc/default/sphinxsearch
-
-- name: Start and Enable sphinxsearch
```
Comment (Ppjet6) [inline ansible/roles/install_sphinx/tasks/main.yml]: Removing this is not enough. By default on Debian services are started after install.

You either need to mask the (non-existing) service, install, unmask, or return 101 from [policy-rc.d](https://people.debian.org/~hmh/invokerc.d-policyrc.d-specification.txt).

I haven't checked what we do exactly in imouto (if we make sure services are started/enabled after install), maybe the policy-rc.d change would break too many assumptions, maybe not.

File: ansible/host_vars/default [hunk 2]
```diff
@@ -47,7 +47,7 @@ code_dir: /home/{{ansible_ssh_user}}/Tatoeba
 # Number of revisions to keep in verions directory (for later restore); should be greater than 0
 revision_limit: 3
 git_temp_dir: .temp_repo
-repo_version: HEAD
+repo_version: dev
```
Comment (Ppjet6) [inline ansible/host_vars/default]:
```
Setting HEAD here makes it pull the 'master' branch, but there is no such branch.
```

Nit: the change itself is harmless, and probably more explicit. I am mostly curious about commit message. github.com/tatoeba/tatoeba2 has `dev` as default branch, I wonder why `master` would be pulled by default.

File: Vagrantfile [hunk 3]
```diff
@@ -7,6 +7,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.box = "debian/stretch64"
   config.vm.network :forwarded_port, guest: 80, host: 8080
   config.vm.synced_folder '.', '/vagrant', disabled: true
+
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 2048
```
Comment (Ppjet6) [inline Vagrantfile]: Should we not reduce the number of language indexed in dev instead? Do we need all of them all the time?
It might be useful to figure out corner cases here and there but that could be part of a test suite maybe?

**Comment by Ppjet6** [comment]
Thanks! That's a lot of changes in there.
I fought the review system but I failed, you might find a few comments here and there.

I also push [patches](https://github.com/Ppjet6/imouto/commits/stretch-service-file) for service files for tatodetect, sinoparserd and suggestd, if you want to have a look.

I have one concern, but it might not be specific to this PR so you might want to discard it, is that it seems we're only thinking about the bring-up bits. I am not sure what would happen if we used the same playbook against a host twice.

File: Vagrantfile (hunk 3 above)
Comment (jiru) [reply Vagrantfile]: I agree. Increasing RAM here was more of a temporary solution to get things working. I opened a ticket for that: Tatoeba/tatoeba2#1689
